image-loader/app/model/Uploader.scala

```diff
@@ -0,0 +1,340 @@
+             optimiseOps: OptimiseOps,
+             browserViewableImage: StorableBrowserViewableImage,
+             optimisedFileMetadata: FileMetadata)
+           (implicit ec: ExecutionContext, logMarker: LogMarker): Future[Option[StorableOptimisedImage]] = {
+    if (optimiseOps.shouldOptimise(Some(browserViewableImage.mimeType), optimisedFileMetadata)) {
+      for {
+        (optimisedFile: File, optimisedMimeType: MimeType) <- optimiseOps.toOptimisedFile(browserViewableImage.file, browserViewableImage, tempDir)
+      } yield Some(browserViewableImage.copy(file = optimisedFile).copy(mimeType = optimisedMimeType).asStorableOptimisedImage)
+    } else if (browserViewableImage.mustUpload) {
+      Future.successful(Some(browserViewableImage.asStorableOptimisedImage))
+    } else
+      Future.successful(None)
+  }
+
+  def toMetaMap(uploadRequest: UploadRequest): Map[String, String] = {
+    val baseMeta = Map(
+      "uploaded_by" -> uploadRequest.uploadedBy,
+      "upload_time" -> printDateTime(uploadRequest.uploadTime)
+    ) ++ uploadRequest.identifiersMeta
+
+    uploadRequest.uploadInfo.filename match {
+      case Some(f) => baseMeta ++ Map("file_name" -> URLEncoder.encode(f, StandardCharsets.UTF_8.name()))
+      case _ => baseMeta
+    }
+  }
+
+  private def toFileMetadata(f: File, imageId: String, mimeType: Option[MimeType]): Future[FileMetadata] = {
+    mimeType match {
+      case Some(Png | Tiff) => FileMetadataReader.fromIPTCHeadersWithColorInfo(f, imageId, mimeType.get)
+      case _ => FileMetadataReader.fromIPTCHeaders(f, imageId)
+    }
+  }
+
+  private def createThumbFuture(fileMetadata: FileMetadata,
+                                colourModelFuture: Future[Option[String]],
+                                storableBrowserViewableImage: StorableBrowserViewableImage,
+                                deps: ImageUploadOpsDependencies)(implicit ec: ExecutionContext) = {
+    import deps._
+    for {
+      colourModel <- colourModelFuture
+      iccColourSpace = FileMetadataHelper.normalisedIccColourSpace(fileMetadata)
+      (thumb, thumbMimeType) <- imageOps
+        .createThumbnail(storableBrowserViewableImage.file, Some(storableBrowserViewableImage.mimeType), config.thumbWidth,
+          config.thumbQuality, config.tempDir, iccColourSpace, colourModel)
+    } yield storableBrowserViewableImage
+      .copy(file = thumb, mimeType = thumbMimeType)
+      .asStorableThumbImage
+  }
+
+  private def createBrowserViewableFileFuture(uploadRequest: UploadRequest,
+                                              tempDir: File,
+                                        deps: ImageUploadOpsDependencies)(implicit ec: ExecutionContext): Future[StorableBrowserViewableImage] = {
+    import deps._
+    uploadRequest.mimeType match {
+      case Some(mime) if config.transcodedMimeTypes.contains(mime) =>
+        for {
+          (file, filetype) <- imageOps.transformImage(uploadRequest.tempFile, uploadRequest.mimeType, tempDir)
```

Comment:
strictly speaking `fileType` here is the extension of the file, which presumably can be tif, tiff, TIF, TIFF etc....